lib/trace.js

```diff
@@ -283,7 +283,7 @@ const traceFile = async ({
       // Remove empty names (e.g., from allowed missing modules).
       .filter(Boolean)
       // Remove core standard Node.js libraries (`path`, etc.)
-      .filter(({ depPath }, i) => depPath !== depNames[i])
+      .filter(({ depPath }) => depNames.indexOf(depPath) !== -1)
```

Comment:
If performance is a concern (O(n)), instead of returning null when isAllowed is true, you might return an object containing depPath in line 257 instead of null. Then you can keep depNames[I] as is, which would be O(1).